examples/provider/provider_env.tf

```
@@ -0,0 +1,11 @@
+terraform {
+  required_providers {
+    upcloud = {
+      source  = "UpCloudLtd/upcloud"
+      version = "~> 2.0"
+    }
+  }
+}
+
+provider "upcloud" {
+}
```

Is this example missing some env variables?

---

If it's meant to be like this, then maybe add some comment here to make it clear 🤔 

```suggestion
  # this is left empty intentionally, UpCloud credentials are read from the environment variables UPCLOUD_USERNAME and UPCLOUD_PASSWORD
}
```

---

Those examples are included to `docs/index.md` by tfplugindocs and those code snippets are documented there. But you are right those examples on their own won't make much sense :) I'll add more comments to those tf files also.